### PR TITLE
Updated "Models defined by data"

### LINKED ISSUE #3406: Models defined by data
Is this correct (https://pytorch-lightning.readthedocs.io/_/downloads/en/latest/pdf/ pg17)

The first code example shows `prepare_data()` and `setup()` being called on the dataloader, but then it is not passed to `fit()`?

The second box has no explanation - I assume it's an alternative to the first code example?

image

### PR DESCRIPTION
**Pull request by david-waterworth**
Updated documentation to correct code-block showing use of DataModule

Fixes #3406

### CODE REVIEW

**Comment by david-waterworth** [comment]
Sorry didn't test with -W flag (treat warnings as errors)

File: docs/source/introduction_guide.rst [hunk 1]
```diff
@@ -289,15 +289,15 @@ When your models need to know about the data, it's best to process the data befo
...
 
 
 1. use `prepare_data` to download and process the dataset.
 2. use `setup` to do splits, and build your model internals
 
-|
+An alternative to using a DataModule is to defer initialization of the models modules to the `setup` method of your LightningModule as follows:
 
-.. testcode::
+.. code-block:: python
```
Comment (Borda) [inline docs/source/introduction_guide.rst:300]: any reason why this?

**Comment by david-waterworth** [comment]
Sorry no i changed that when I was looking for why I was getting the warning. That code block wasn’t documented and I have a feeling it was left in from previously - I did add a comment before it to indicate that it was an alternative. I can revert the change if required?

**Comment by Borda** [comment]
> Sorry no i changed that when I was looking for why I was getting the warning. That code block wasn’t documented and I have a feeling it was left in from previously - I did add a comment before it to indicate that it was an alternative. I can revert the change if required?

I would do it... pls
cc: @awaelchli

**Comment by david-waterworth** [comment]
@Borda How do I do this, do I create a new pull request? Since this one has been closed it's not sync'ing from my branch?

**Comment by Borda** [comment]
yes, create a new PR and in such case you may review if with minimal change we run some other examples as a test, mostly just to verify syntax and compatibility with actual API

**Pull request merged**